prototype1/src/app_net/helpers.rs

```diff
@@ -11,7 +11,9 @@ mod helpers {
       .and_then(|external_ip| TcpStream::connect((external_ip, 80)))
       .and_then(|stream| {
         let addr = stream.local_addr();
-        stream.shutdown(Shutdown::Both);
+
+        // We don't care about the tcp connection, just the IP
+        let _ = stream.shutdown(Shutdown::Both);
```

Comment:
why not something we can stick in util and forget about, like
```rust
pub trait IgnoreResult {
    fn ignore_result(self);
}

impl<S, E> IgnoreResult for Result<S, E> {
    fn ignore_result(self) { }
}
```

and then whenever we do side-effect stuff that returns a result we do
```rust
stream.shutdown(Shutdown::Both).ignore_result()
```